3-networks/modules/restricted_shared_vpc/variables.tf

```diff
@@ -0,0 +1,120 @@
+/**
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+variable "org_id" {
+  type        = string
+  description = "Organization ID"
+}
+
+variable "project_id" {
+  type        = string
+  description = "Project ID for Restricted Shared VPC."
+}
+
+variable "project_number" {
+  type        = number
+  description = "Project number for Restricted Shared VPC. It is the project INSIDE the regular service perimeter."
+}
+
+variable "environment_code" {
+  type        = string
+  description = "A short form of the folder level resources (environment) within the Google Cloud organization."
+}
+
+variable "nat_enabled" {
+  type        = bool
+  description = "Toggle creation of NAT cloud router."
+  default     = false
+}
+
+variable "nat_bgp_asn1" {
+  type        = number
+  description = "BGP ASN for first NAT cloud routes."
+  default     = 0
```

Comment:
Should we provide a default value for this? I would also prefer it was region1 so its clear which Cloud Router region / NAT it is associated with

Reply:
NAT is now disable by the default and all NAT related variables have become optional. 
Adding a better description for it.

Reply:
Thanks! So after some extra thought, we could make this one ASN to make it consistent with the way subnet cloud routers are currently being created? Not a blocker for merging.

Reply:
Also I think it would be best to have a default value which is valid